datafusion/optimizer/src/filter_push_down.rs

```diff
@@ -122,6 +122,27 @@ fn remove_filters(
         .collect::<Vec<_>>()
 }
 
+// rename all filter columns which have alias name
+fn rename_filters_column_name(
+    filters: &mut [Predicate],
+    alias_cols_expr_and_name: &HashMap<&Expr, &String>,
+) {
+    for (expr, columns) in filters {
+        if alias_cols_expr_and_name.contains_key(expr) {
+            let col_string = <&std::string::String>::clone(
+                alias_cols_expr_and_name.get(expr).unwrap(),
+            );
```

Comment:
```suggestion
            let col_string = alias_cols_expr_and_name.get(expr).unwrap();
```

Seemed to work for me locally